### PR TITLE
Add Okta service configuration.

### PR DESCRIPTION
**Pull request by mdwn**
Users will be able to specify to Okta service configuration. This allows users to enable the service as well as specify the `api_endpoint` and `api_token_path` for the service.

### CODE REVIEW

File: lib/service/servicecfg/config.go [hunk 1]
```diff
@@ -660,5 +664,5 @@ func verifyEnabledService(cfg *Config) error {
 	}
 
 	return trace.BadParameter(
-		"config: enable at least one of auth_service, ssh_service, proxy_service, app_service, database_service, kubernetes_service, windows_desktop_service or discovery_service")
+		"config: enable at least one of auth_service, ssh_service, proxy_service, app_service, database_service, kubernetes_service, windows_desktop_service, discovery_service, or okta_service")
```
Comment (nklaassen) [inline lib/service/servicecfg/config.go:667]: Ahh, I see you have the correct opinion on the oxford comma, I approve

File: integrations/operator/controllers/resources/suite_test.go [hunk 2]
```diff
@@ -51,7 +51,6 @@ import (
 	"github.com/gravitational/teleport/lib/modules"
 	"github.com/gravitational/teleport/lib/service/servicecfg"
 	"github.com/gravitational/teleport/lib/utils"
-	//+kubebuilder:scaffold:imports
```
Comment (nklaassen) [inline integrations/operator/controllers/resources/suite_test.go:54]: is this related to the PR?

File: integrations/operator/main.go [hunk 3]
```diff
@@ -41,7 +41,6 @@ import (
 	resourcesv5 "github.com/gravitational/teleport/integrations/operator/apis/resources/v5"
 	"github.com/gravitational/teleport/integrations/operator/controllers/resources"
 	sidecar2 "github.com/gravitational/teleport/integrations/operator/sidecar"
-	//+kubebuilder:scaffold:imports
```
Comment (nklaassen) [inline integrations/operator/main.go:44]: same question, is this related to the PR?

File: lib/service/servicecfg/config_test.go [hunk 4]
```diff
@@ -533,3 +534,106 @@ func TestValidateConfig(t *testing.T) {
...
+	}{
+		{
+			desc: "auth enabled",
+			config: &Config{
+				Auth: AuthConfig{
+					Enabled: true,
+				},
+			},
+			errAssertionFunc: require.NoError,
+		},
```
Comment (nklaassen) [inline lib/service/servicecfg/config_test.go:552]: This seems super verbose for this simple logic it's testing, I'm tempted to suggest something like the following if it works but it's probably way less readable so I'll leave it up to you

```go
cfg := &Config{}
tests := []struct{
	name string
	enabled *bool
}{
	{"auth", &cfg.Auth.Enabled},
	{"ssh", &cfg.SSH.Enabled},
	...
}
```

File: lib/config/configuration.go [hunk 5]
```diff
@@ -2267,3 +2275,30 @@ func applyTokenConfig(fc *FileConfig, cfg *servicecfg.Config) error {
...
+	if fc.Okta.APITokenPath == "" {
+		return trace.BadParameter("okta_service is enabled but no api_token_path is specified")
+	}
+
+	// Make sure the URL is valid
+	url, err := url.Parse(fc.Okta.APIEndpoint)
+	if err != nil {
+		return trace.NewAggregate(trace.BadParameter("malformed URL %s", fc.Okta.APIEndpoint), err)
+	}
+	fmt.Printf("%s\n", url)
```
Comment (tigrato) [inline lib/config/configuration.go:2292]: debug data

File: lib/config/configuration.go [hunk 6]
```diff
@@ -2267,3 +2275,30 @@ func applyTokenConfig(fc *FileConfig, cfg *servicecfg.Config) error {
...
+
+	// Make sure the API token exists.
+	_, err = utils.StatFile(fc.Okta.APITokenPath)
+	if err != nil {
```
Comment (tigrato) [inline lib/config/configuration.go:2296]:
```suggestion
	
	if _, err = utils.StatFile(fc.Okta.APITokenPath); err != nil {
```

File: lib/config/configuration.go [hunk 7]
```diff
@@ -2267,3 +2275,30 @@ func applyTokenConfig(fc *FileConfig, cfg *servicecfg.Config) error {
...
+func applyOktaConfig(fc *FileConfig, cfg *servicecfg.Config) error {
+	if fc.Okta.APIEndpoint == "" {
+		return trace.BadParameter("okta_service is enabled but no api_endpoint is specified")
+	}
+	if fc.Okta.APITokenPath == "" {
+		return trace.BadParameter("okta_service is enabled but no api_token_path is specified")
+	}
+
+	// Make sure the URL is valid
+	url, err := url.Parse(fc.Okta.APIEndpoint)
```
Comment (tigrato) [inline lib/config/configuration.go:2288]: `url.Parse` can return nil if no Host/schema is provided.
It's probably better to check them as well if required

File: integrations/operator/controllers/resources/suite_test.go (hunk 2 above)
Comment (mdwn) [reply integrations/operator/controllers/resources/suite_test.go:54]: Oh huh, no. This got in by accident!

File: integrations/operator/main.go (hunk 3 above)
Comment (mdwn) [reply integrations/operator/main.go:44]: Same!

File: lib/service/servicecfg/config_test.go (hunk 4 above)
Comment (mdwn) [reply lib/service/servicecfg/config_test.go:552]: How about I one-line all of these instead? That way it doesn't seem so overly dense.

**Review by nklaassen — APPROVED** [review]

**Review by tigrato — APPROVED** [review]
